flow-server/src/main/java/com/vaadin/flow/server/communication/rpc/PublishedServerEventHandlerRpcHandler.java

```
@@ -58,28 +68,307 @@ public String getRpcType() {
+                                    + "a 'null' value was received for parameter %d "
+                                    + "which refers to primitive type '%s' "
+                                    + "in the method '%s' defined in the class '%s'",
+                            index, type.getName(), method.getName(),
+                            method.getDeclaringClass().getName());
+            throw new IllegalArgumentException(msg);
+        } else if (type.isArray()) {
+            return decodeArray(method, type, index, argValue);
+        } else {
+            Class<?> convertedType = ReflectTools.convertPrimitiveType(type);
+
+            try {
+                final Class<?> polymerUtil = PublishedServerEventHandlerRpcHandler.class
+                        .getClassLoader().loadClass(
+                                "com.vaadin.flow.component.polymertemplate.rpc.PublishedEventRpcHandlerUtil");
+                if ((Boolean) polymerUtil
+                        .getMethod("isTemplateModelValue", Component.class,
+                                JsonValue.class, Class.class)
+                        .invoke(polymerUtil, instance, argValue,
+                                convertedType)) {
+                    return polymerUtil
+                            .getMethod("getTemplateItem", Component.class,
+                                    JsonObject.class, Type.class)
+                            .invoke(polymerUtil, instance, argValue,
+                                    method.getGenericParameterTypes()[index]);
+                }
+            } catch (ClassNotFoundException | NoSuchMethodException e) {
+                // NO-OP we just try to check if the polymer library is added
+                LoggerFactory
+                        .getLogger(PublishedServerEventHandlerRpcHandler.class)
+                        .trace("Tried finding polymer dependency classes.", e);
+            } catch (IllegalAccessException | InvocationTargetException e) {
```

`IllegalAccessException` : can't happen. If it happens then this is an impl mistake and it should throw probably.

`InvocationTargetException`  means there was some exception in calling the method (which is wrapped to this one because of reflection call).
This exception may not be silently ignored (or just logged). It should be thrown somehow.